### PR TITLE
Added Formik Recipe

### PR DESCRIPTION
**Pull request by luistak**
**What** 

Added a basic async Formik example test recipe based in [this example](https://formik.org/docs/examples/async-submission)  as requested in #536

### CODE REVIEW

**Comment by luistak** [comment]
Is this PR good to go?

File: website/sidebars.json [hunk 1]
```diff
@@ -133,6 +133,7 @@
       "example-findByText",
       "example-input-event",
       "example-react-context",
+      "example-react-formik",
```
Comment (nickmccurdy) [inline website/sidebars.json:136]: If we move this after`useReducer`, the React core examples will be grouped before the third party library examples.

File: docs/example-formik.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+```jsx
+// myForm.test.js
+import React from 'react'
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import { MyForm } from './myForm.js'
+
+test('rendering and submiting a basic Formik form', async () => {
+  const mockedOnSubmit = jest.fn()
```
Comment (nickmccurdy) [inline docs/example-formik.md:64]: Let's name this `handleSubmit` to follow React's naming convention for matched `on*` and `handle*` props.

**Review by nickmccurdy — CHANGES_REQUESTED** [review]

**Comment by luistak** [comment]
Hey thank you @nickmccurdy I was kinda busy these days I'm glad you fixed those comments ❤️

**Pull request merged**